Sources/ConfidenceProvider/ConfidenceFeatureProvider.swift

```diff
@@ -48,7 +48,7 @@ public class ConfidenceFeatureProvider: FeatureProvider {
     }
 
     /// Initialize the Provider via a `Confidence` object.
-    public init(confidence: Confidence) {
+    public init(confidence: Confidence, session: URLSession? = nil) {
```

Comment:
I was thinking we could have all the configuration and injectables in the Confidence.Builder level. Here in the Provider's init the user simply passes the built Confidence object. Wdyt?

Reply:
so the session in this case can be exposed from the confidence object? since the session it's used to initialize the client like this:
```
let client = RemoteConfidenceResolveClient(
                options: options,
                session: self.session,
                applyOnResolve: false,
                flagApplier: flagApplier,
                metadata: metadata
            )
```

Reply:
Mmh exposing that from `Confidence` doesn't sound great either, actually. Could we have a private init at least here to not expose `session` to the final users? I just think that keeping this simple would be nice:
```
 /// Initialize the Provider via a `Confidence` object.
    public init(confidence: Confidence) {
```